### PR TITLE
Parse and register schema declarations lazily

### PR DESCRIPTION
**Pull request by zdevito**
This should prevent slow startup times, and will not report as many
errors during static initialization time which are hard to debug

@ezyang @apaszke

### CODE REVIEW

**Comment by zdevito** [comment]
#9760

File: torch/csrc/jit/script/compiler.cpp [hunk 1]
```diff
@@ -373,13 +373,16 @@ static bool isTensorSubtype(Value* v) {
 at::optional<std::vector<int64_t>> getIntListAttribute(at::optional<int32_t> N, Value* input) {
   auto list = constant_as<Shared<jit::IntList>>(input);
   if(list)
-    return std::vector<int64_t>(*list);
+    return std::vector<int64_t>(list.value()->elements());
```
Comment (zdevito) [inline torch/csrc/jit/script/compiler.cpp]: This is a one liner bug fix.

File: torch/csrc/jit/operator.cpp [hunk 2]
```diff
@@ -260,18 +264,27 @@ struct OperatorRegistry  {
...
-    operators_by_sig[canonicalSchemaString(op.schema)] = op_ptr;
+public:
+  void registerOperator(Operator&& op) {
+    std::lock_guard<std::mutex> guard(lock);
+    to_register.push_back(std::make_shared<Operator>(std::move(op)));
   }
 
+
   Operator& lookupByLiteral(const char * name) {
+    std::lock_guard<std::mutex> guard(lock);
```
Comment (zdevito) [inline torch/csrc/jit/operator.cpp]: This lock was missing eventhough operators_by_sig_literal was being written to in this functions. I doubt an uncontended lock will add much overhead here.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**